cloudberry/zion/src/main/scala/edu/uci/ics/cloudberry/zion/model/impl/JSONParser.scala

```diff
@@ -56,6 +56,41 @@ object JSONParser {
     resolved
   }
 
+  /**
+    * Parses json request and extract [[QueryExeOption]], then removes all option-related field from requests.
+    *
+    * @param json
+    * @return
+    */
+  def extractQueryExeOption(json: JsValue): (JsValue, QueryExeOption) = {
```

Comment:
the logic of this function seems hard to follow. Could you elaborate the logic in the ScalaDoc? 
And if the logic is more appropriate in the `if else` block, we'd better write the logic in that way than the `case` way.

Reply:
The parsing logic is
- if there is no `option`, treat queries by the original way (meaning if there is `limit` in the query, it will be executed in database statement not controlled by cloudberry, just as it is today)

- if there is `option`,

  - for `batch` queries, if `limit` presents, throw errors. Because now twittermap use `limit` in `batch` (the point we were missing in the previous discussion), there is one specialization for this: we enable `aggregate` and `limit` present together.

  - for single query, if `limit` presents, this is the case we are targeting. We allow two possible APIs.
```
{
    "select" : {
        "limit": 100
    }
}
```
and 
```
{
    "select" : {
        "order" : [ "-create_at"],  // required
        "limit": 100,                       // required
        "offset" : 0,                       // required
        "field": ["create_at", "id"] // optional
    }
}
```
for the first one, we remove `select` field after parsing the `limit` value to `QueryExeOption` and for the second one we take the limit value and reassign `Int.MaxValue` to `limit`. This way we keep the format constrains in `select` statement and parse the `limit` as well.